client/src/app/pages/controls/tags/tags.tsx

```diff
@@ -157,281 +152,283 @@ export const Tags: React.FC = () => {
-        return colorLabel || hex;
-      },
-    },
-  ];
-  const { filterValues, setFilterValues, filteredItems } = useLegacyFilterState(
-    tagCategories || [],
-    filterCategories
-  );
-
-  const getSortValues = (item: TagCategory) => [
-    "",
-    item?.name || "",
-    typeof item?.rank === "number" ? item.rank : Number.MAX_VALUE,
-
-    "",
-    item?.tags?.length || 0,
-    "", // Action column
-  ];
-
-  const { sortBy, onSort, sortedItems } = useLegacySortState(
-    filteredItems,
-    getSortValues
-  );
-
-  const { currentPageItems, setPageNumber, paginationProps } =
-    useLegacyPaginationState(sortedItems, 10);
-
-  const deleteTagFromTable = (row: Tag) => {
-    setTagToDelete(row);
+  const deleteTagFromTable = (tag: Tag) => {
+    setTagToDelete(tag);
   };
 
-  const columns: ICell[] = [
-    {
-      title: t("terms.tagCategory"),
-      transforms: [sortable],
-      cellFormatters: [expandable],
-    },
-    { title: t("terms.rank"), transforms: [sortable] },
-    {
-      title: t("terms.color"),
-      transforms: [],
-    },
-    {
-      title: t("terms.tagCount"),
-      transforms: [sortable],
+  const tableControls = useLocalTableControls({
+    tableName: "business-services-table",
+    idProperty: "name",
+    items: tagCategories,
+    columnNames: {
+      name: t("terms.name"),
+      rank: t("terms.rank"),
+      color: t("terms.color"),
+      tagCount: t("terms.tagCount"),
     },
-    {
-      title: "",
-      props: {
-        className: "pf-v5-u-text-align-right",
-      },
-    },
-  ];
-
-  const rows: IRow[] = [];
-  currentPageItems.forEach((item) => {
-    const isExpanded = isItemExpanded(item) && !!item?.tags?.length;
-    const categoryColor = item.colour || getTagCategoryFallbackColor(item);
-    rows.push({
-      [ENTITY_FIELD]: item,
-      isOpen: (item.tags || []).length > 0 ? isExpanded : undefined,
-      cells: [
-        {
-          title: item.name,
-        },
-        {
-          title: item.rank,
+    isFilterEnabled: true,
+    isSortEnabled: true,
+    isPaginationEnabled: true,
+    hasActionsColumn: true,
+    isExpansionEnabled: true,
+    expandableVariant: "single",
+    filterCategories: [
+      {
+        categoryKey: "tags",
+        title: t("terms.name"),
+        type: FilterType.multiselect,
+        placeholderText:
+          t("actions.filterBy", {
+            what: t("terms.name").toLowerCase(),
+          }) + "...",
+        getItemValue: (item) => {
+          const tagCategoryNames = item.name?.toString() || "";
+          const tagNames = item?.tags
+            ?.map((tag) => tag.name)
+            .concat(tagCategoryNames)
+            .join("");
+          return tagNames || "";
         },
-        {
-          title: <Color hex={categoryColor} />,
-        },
-        {
-          title: item.tags ? item.tags.length : 0,
+        selectOptions: dedupeFunction(
+          tagCategories
+            ?.map((tagCategory) => tagCategory?.tags)
```

Comment:
1. if we suspect that the array is nullish then we should continue with the `?.` operator till the end
2. how about:  ` tagCategory?.tags ?? []`
3. `flatMap`?